include/deal.II/fe/fe_tools.h

```diff
@@ -909,12 +909,23 @@ namespace FETools
    * correct size, which is equal to the number of degrees of freedom in the
    * finite element.
    */
-
   template <int dim>
   void
   hierarchic_to_lexicographic_numbering(unsigned int               degree,
                                         std::vector<unsigned int> &h2l);
 
+  /**
+   * Like the previous function but instead of returning its result through
+   * the last argument return it as a value.
+   */
+  template <int dim>
+  std::vector<unsigned int>
+  hierarchic_to_lexicographic_numbering(unsigned int degree);
```

Comment:
Could we deprecate the corresponding older functions?

Reply:
Yes, that is a good idea; in my opinion we should deprecate all functions but the two using the `degree` argument and returning a vector, so also the ones taking `FiniteElementData`. These functions are mostly used within the library and one can always easily get the degree from the finite element data, so I would just keep the simplest form possible. Let me add that as a commit to this PR.